index.js

```diff
@@ -58,29 +68,79 @@ module.exports = (input, options) => {
   let table = ''
 
   // header line
-  table += row(titles.map(
-    (title, i) => pad(alignments[i], widths[i], title)
-  ))
+  table += row(alignments, widths, titles, gutters)
 
   // header separator
-  table += row(alignments.map(
+  table += line(alignments.map(
     (align, i) => (
       (align === 'LEFT' || align === 'CENTER' ? ':' : '-') +
       repeat('-', widths[i] - 2) +
       (align === 'RIGHT' || align === 'CENTER' ? ':' : '-')
     )
-  ))
+  ), true)
 
   // table body
   table += input.map(
-    item => row(keys.map(
-      (key, i) => pad(alignments[i], widths[i], stringify(item[key]))
-    ))
+    (item, i) => row(alignments, widths, keys.map(
+      key => stringify(item[key])
+    ), gutters)
   ).join('')
 
   return table
 }
 
+function row (alignments, widths, columns, gutters) {
+  const width = columns.length
+  const values = new Array(width)
+  const first = new Array(width)
+  let height = 1
+
+  for (let h = 0; h < width; h++) {
```

Comment:
Let's compress a bit from here to line 110, the empty lines here remind me of double spacing papers for essays back in the day :laughing:

```js
  for (let h = 0; h < width; h++) {
    const cells = values[h] = split(columns[h], widths[h])
    if (cells.length > height) height = cells.length
    first[h] = pad(alignments[h], widths[h], cells[0])
  }
 
  if (height === 1) return line(first, true)
 
  const lines = new Array(height)
  lines[0] = line(first, true)
```

Reply:
Gotcha.

Reply:
Gotcha.